apis/v1alpha2/policy_types.go

```diff
@@ -38,6 +38,15 @@ type PolicyTargetReference struct {
 	//
 	// +optional
 	Namespace *Namespace `json:"namespace,omitempty"`
+
+	// SectionName is the name of a section within the target resource. When
+	// unspecified, this targetRef targets the entire resource. In the following
+	// resources, SectionName is interpreted as the following:
+	// * Gateway: Listener Name
+	// * Service: Port Name
+	//
```

Comment:
On further review, I think we need to be specific here about what happens if this is not found.

Would you be able to add some wording like:
```go
	// 
	// If a SectionName is specified, but does not exist on the targeted object,
	// the Policy must fail to attach, and the policy implementation should record
	// a `ResolvedRefs` or similar Condition in the Policy's status.
```

(We should specify something similar at the top level as well, but that's another issue).